### PR TITLE
feat: Add parameter `number_of_tree` to `RandomForest` classifier and regressor

### LINKED ISSUE #161: Set number of trees for random forests
### Is your feature request related to a problem?

It's not possible to set the number of trees of random forests (classifier & regressor).

### Desired solution

* Add a new parameter `number_of_trees: int` to the initializer of `safeds.ml.classification.RandomForest` and `safeds.ml.regression.RandomForest`
* Raise a value error if `number_of_trees` < 1
* Pass it along to `n_estimators` of the wrapped `scikit-learn` model in the `fit` method

### Possible alternatives (optional)

_No response_

### Screenshots (optional)

_No response_

### Additional Context (optional)

_No response_

### PR DESCRIPTION
**Pull request by alex-senger**
Closes #161.

### Summary of Changes

Added number_of_trees parameter to initiator of random_forest_classifier and random_forest_regressor.

### CODE REVIEW

**Comment by lars-reimann** [comment]
## [🦙 MegaLinter](https://megalinter.io/6.22.1) status: ❌ [ERROR](https://github.com/Safe-DS/Stdlib/actions/runs/4764408615)

| Descriptor  |                                 Linter                                 |Files|Fixed|                            Errors                            |Elapsed time|
|-------------|------------------------------------------------------------------------|-----|----:|--------------------------------------------------------------|------------|
|✅ PYTHON    |[black](https://megalinter.io/6.22.1/descriptors/python_black)          |    4|    2|                                                             0|0.6s        |
|❌ PYTHON    |[mypy](https://megalinter.io/6.22.1/descriptors/python_mypy)            |    4|     |[2](https://github.com/Safe-DS/Stdlib/actions/runs/4764408615)|1.79s       |
|❌ PYTHON    |[ruff](https://megalinter.io/6.22.1/descriptors/python_ruff)            |    4|    2|[4](https://github.com/Safe-DS/Stdlib/actions/runs/4764408615)|0.04s       |
|✅ REPOSITORY|[git_diff](https://megalinter.io/6.22.1/descriptors/repository_git_diff)|yes  |     |no                                                            |0.02s       |

See detailed report in [MegaLinter reports](https://github.com/Safe-DS/Stdlib/actions/runs/4764408615)
_Set `VALIDATE_ALL_CODEBASE: true` in mega-linter.yml to validate all sources, not only the diff_

_MegaLinter is graciously provided by [OX Security](https://www.ox.security/?ref=megalinter)_

**Review by PhilipGutberlet — APPROVED** [review]

File: src/safeds/ml/classical/classification/_random_forest.py [hunk 1]
```diff
@@ -13,9 +13,23 @@
...
+
+    Parameters
+    ----------
+    number_of_trees : int
+        The number of trees to be used in the random forest. Has to be greater than 0.
+
+    Raises
+    ------
+    ValueError
+        If the number of trees is less than 1.
```
Comment (Marsmaennchen221) [inline src/safeds/ml/classical/classification/_random_forest.py:26]:
```suggestion
        If the number of trees is less than or equal to 0.
```
I think it is better to understand if you use the same value in the docs (required greater than 0 -Error if less than or equal to 0)

File: src/safeds/ml/classical/regression/_random_forest.py [hunk 2]
```diff
@@ -13,9 +13,23 @@
...
+
+    Parameters
+    ----------
+    number_of_trees : int
+        The number of trees to be used in the random forest. Has to be greater than 0.
+
+    Raises
+    ------
+    ValueError
+        If the number of trees is less than 1.
```
Comment (Marsmaennchen221) [inline src/safeds/ml/classical/regression/_random_forest.py:26]:
```suggestion
        If the number of trees is less than or equal to 0.
```
See Classification

**Review by Marsmaennchen221 — CHANGES_REQUESTED** [review]

**Review by lars-reimann — APPROVED** [review]

**Pull request merged**